datameta/api/metadatasets.py

```diff
@@ -192,10 +192,21 @@ def post(request: Request) -> MetaDataSetResponse:
         ]
 
         missing_msets = [("Invalid metadataset id.", mset_id) for mset_id, target_mset in msets if target_mset is None]
+
         if missing_msets:
             messages, entities = zip(*missing_msets)
             raise errors.get_validation_error(messages=messages, entities=entities)
 
+        already_replaced = [
+            (f"MetaDataSet was already replaced via event {target_mset.replaced_via_event_id}", mset_id)
```

Comment:
This exposes an internal database ID. Also, I'd hide the event logic to the user and rather print the ID of the new mset (`get_identifier`).